### ISSUE TITLE
No scrollbars in RecyclerView

### ISSUE THREAD

**Issue by nikclayton** [tuskyapp/Tusky#3537]
The app doesn't show scrollbars on `RecyclerView`.

Why this is not the default behaviour I don't know. The fix is to add

```
android:scrollbars="vertical"
```

to every `RecyclerView` in a layout XML file.

https://wandering.shop/@kagan/109904256200987475 is an example user complaint about this.

* * * *
- Tusky Version: 21

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by Lakoja** [comment]
Is that configurable (in Android)?

That option has no effect on my timeline (scrollbar is still hidden after scrolling).
The one that is visible is anyway only a tiny dot because the home timeline is so long.

And auto-hidden and too small scrollbars are the norm (a general "ui feature or decision" if you will). I see that everywhere.
Or conversely I have never seen an app or mobile webpage that has other scrollbars.

**Comment by nikclayton** [comment]
> Is that configurable (in Android)?

Yes, it's configurable on Android, hence the `android:scrollbars` attribute.

> That option has no effect on my timeline (scrollbar is still hidden after scrolling).

The attribute doesn't control what happens when you *stop* scrolling, only what happens if you are scrolling.

Try this:

1. Scroll up and down the home timeline. You will see no scrollbar while you are scrolling.
2. Open a thread with enough messages that you need to scroll, and scroll up and down. You will see a scrollbar while you are scrolling, and it will fade out shortly after you stop scrolling

This is because `fragment_view_thread.xml` sets the `android:scrollbar` attribute and `fragment_timeline.xml` does not.

The fadeout / hiding of the scrollbar when you stop scrolling can be disabled by setting `android:fadeScrollbars="false"`, also on the `RecyclerView`.

>  The one that is visible is anyway only a tiny dot because the home timeline is so long.

That's controllable by the `android:scrollbar*` attributes common to everything that inherits from `View` (https://developer.android.com/reference/android/view/View#xml-attributes)

>  Or conversely I have never seen an app or mobile webpage that has other scrollbars.

Not sure what you mean by "other scrollbars" here.

**Comment by Lakoja** [comment]
>     1. Scroll up and down the home timeline. You will see no scrollbar while you are scrolling.

You are right.
I never missed it there as a normal Android scrollbar is so slim that it is not usable or informative anyway.

**Comment by connyduck** [comment]
Scrollbars are intentionally not shown because with unknown list length and variying item size they are pretty inacurrate and jump around a lot.

**Comment by nikclayton** [comment]
That's true, but they still give the user a rough indication of where they are in the timeline (beginning / middle / end), and would allow them to make a more informed choice about whether to continue reading (they can see that they're almost at one end), or take a break.

To deal with items of varying size the default implementation varies the size of the scrollbar thumb. It's possible to go the other way, and keep the scrollbar thumb a fixed size and adjust the position.

E.g.,

1. Put https://gist.github.com/nikclayton/6eddadbb864e1e541a91f6018fd8d803 in the `.../view` directory
2. Change `TimelineFragment.kt`, around line 401 from:

```kotlin
        binding.recyclerView.layoutManager = LinearLayoutManager(context)
```

   to 

```kotlin
        binding.recyclerView.layoutManager = FixedScrollThumbLinearLayoutManager(context)
```
3. Add `android:scrollbars="vertical"` to the `RecyclerView` in `fragment_timeline.xml`.

**Comment by connyduck** [comment]
I still don't think we should do this
